Google.PowerShell/Logging/GcLogSinkCmdlets.cs

```diff
@@ -93,4 +95,141 @@ protected override void ProcessRecord()
             }
         }
     }
+
+    [Cmdlet(VerbsCommon.New, "GcLogSink")]
+    public class NewGcLogSinkCmdlet : GcLogEntryCmdletWithLogFilter
+    {
+        private class ParameterSetNames
+        {
+            public const string BigQueryDataSetDestination = "BigQueryDataSetDestination";
+            public const string GcsBucketDestination = "GcsBucketDestination";
+            public const string PubSubTopicDestination = "PubSubTopicDestination";
+        }
+
+        /// <summary>
+        /// Enum of version format for log entry.
+        /// See https://cloud.google.com/logging/docs/api/reference/rest/v2/organizations.sinks#versionformat.
+        /// </summary>
+        public enum LogEntryVersionFormat
+        {
+            /// <summary>
+            /// LogEntry version 2 format.
+            /// </summary>
+            V2,
+
+            /// <summary>
+            /// LogEntry version 1 format.
+            /// </summary>
+            V1
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to check for log entries. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public LogEntryVersionFormat? OutputVersionFormat { get; set; }
```

Comment:
These are all public and should have comments, especially if they are user visivble